### PR TITLE
Add assert_not_regexp_match

### PR DESCRIPTION
**Pull request by zcorpan**
By request from @mikewest 

TODO docs, tests

### CODE REVIEW

File: testharness.js [hunk 1]
```diff
@@ -1077,6 +1077,17 @@ policies and contribution forms [3].
     }
     expose(assert_regexp_match, "assert_regexp_match");
 
+    function assert_not_regexp_match(actual, expected, description) {
+        /*
+         * Test if a string (actual) does not match a regexp (expected)
+         */
+        assert(expected.test(actual) === false,
+               "assert_not_regexp_match", description,
+               "got disallowed value ${actual}",
```
Comment (mikewest) [inline testharness.js]: Nit: Perhaps "${actual} should not have matched ${expected}" would be clearer?

**Comment by mikewest** [comment]
LGTM % a phrasing nit. Thanks!

**Comment by jgraham** [comment]
I am slightly against this in principle; I think there are only a small number of cases where the right thing to do is to allow almost any input to pass except as small set of inputs that fail. It's not too difficult to write this out by hand in the few cases where it's really needed.

**Comment by zcorpan** [comment]
OK, closing

**Pull request closed without merging**